workflow/Snakefile

```diff
@@ -35,6 +38,21 @@ def all_input(wildcards):
             )
         )
 
+    # mapping and merging bam-files
+    wanted_input.extend(
+        expand (
+            [
+                "results/mapped/{sample}-{unit}.bam",
+                "results/mapped/dedup/{sample}-{unit}.bam",
+                "results/mapped/dedup/{sample}-{unit}.metrics.txt",
+                "results/mapped/merged_with_dups.bam",
+                "results/mapped/merged.bam"
```

Comment:
Here, I think it is enough to request the output of the final step. With the suggestions from below, this would be:
```
                "results/merged/{sample}.bam"
```